trustfall_core/src/ir/mod.rs

```diff
@@ -49,8 +54,12 @@ pub struct EdgeParameters(
     #[serde(default, skip_serializing_if = "BTreeMap::is_empty")] pub BTreeMap<Arc<str>, FieldValue>,
 );
 
+/// IR of components of a query, containing information about the vertex ID
+/// of the root of the query, as well as well as maps of all vertices, edges,
+/// folds, and outputs of the query.
```

Comment:
```suggestion
/// A complete component of a query; may itself contain one or more components.
///
/// Contains information about the Vid where the component is rooted,
/// as well as well as maps of all vertices, edges, folds, and outputs from this component.
```
What do you think?
- Is the "one-liner explanation at the top" a good pattern, both for docs.rs and for IDEs?
- I added a bit more explanation to differentiate how query component and query are different.

Reply:
Looking at some `std` docs, it seems to be pretty standard? I do it because when you then look in the lists of for example structs (like [here](https://doc.rust-lang.org/std/sync/index.html#structs)), you get the same one-liner as a brief explanation, so you can see _Ooh that's what I want!_.

I suggest we follow this pattern (which will mean no periods at the end of the one-liner. What do you think?

Also thanks for the explanation!